examples/large_models/tp_llama/llama-handler.py

```diff
@@ -0,0 +1,186 @@
+import logging
+import time
+from abc import ABC
+import json
+import os
+import sys
+import importlib.util
+
+import packaging.version
+import requests
+import torch
+import transformers
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from ts.torch_handler.base_handler import BaseHandler
+current_working_directory = os.getcwd()
+sys.path.insert(0,current_working_directory)
+from llama2 import Llama
+from generate import chat_completion, text_completion, Dialog
+
+logger = logging.getLogger(__name__)
+logger.info("Transformers version %s", transformers.__version__)
+if packaging.version.parse(torch.__version__) >= packaging.version.parse("2.0.0"):
+    logger.info("PyTorch version is 2.0.0 or greater")
+else:
+    logger.info(
+        "PyTorch version is less than 2.0.0, initializing with meta device needs PyTorch 2.0.0 and greater"
+    )
+
+
+
+class LlamaHandler(BaseHandler,ABC):
+    """
+    Transformers handler class for sequence, token classification and question answering.
+    """
+
+    def __init__(self):
+        super(LlamaHandler, self).__init__()
+        self.initialized = False
+
+    def initialize(self, ctx):
+        """
+        In this initialize function, the llama model is loaded using Fairscale and
+        partitioned into multiple stages each on one device using PiPPy.
+        Args:
+            ctx (context): It is a JSON Object containing information
+            pertaining to the model artefacts parameters.
+        """
+        # super().initialize(ctx)
+        if not torch.distributed.is_initialized():
+            torch.distributed.init_process_group("nccl")
+
+        self.manifest = ctx.manifest
+        properties = ctx.system_properties
+        model_dir = properties.get("model_dir")
+        
+        self.device = torch.device(
+            "cuda:" + str(properties.get("gpu_id"))
+            if torch.cuda.is_available() and properties.get("gpu_id") is not None
+            else "cpu"
+        )
```

Comment:
should device be got from rank if torchrun is started?

Reply:
removed the device as we are not using it here, tex_completion or chat_completion take care of it.